redash/handlers/queries.py

```diff
@@ -264,3 +265,23 @@ def post(self, query_id):
         parameter_values = collect_parameters_from_request(request.args)
 
         return run_query(query.data_source, parameter_values, query.query_text, query.id)
+
+
+class QueryVersionListResource(BaseResource):
+    @require_permission('view_query')
+    def get(self, query_id):
+        results = models.Change.query.filter(
```

Comment:
This logic (finding changes for an object) should be in the model (i.e. a method like `Change.for` that takes the model).

Also you need to fetch the query first & check for proper permissions. You can see [an example in alerts](https://github.com/getredash/redash/blob/master/redash/handlers/alerts.py#L47-L49).